apps/vaporgui/FlowEventRouter.cpp

```diff
@@ -1,137 +1,144 @@
-#ifdef WIN32
-//Annoying unreferenced formal parameter warning
-#pragma warning( disable : 4100 )
-#endif
-
-#include <vapor/glutil.h>
-#include <qlineedit.h>
-#include <qscrollarea.h>
-#include <qcolordialog.h>
-#include <QFileDialog>
-#include <vector>
-#include <string>
-#include "vapor/FlowParams.h"
-#include "VariablesWidget.h"
 #include "FlowEventRouter.h"
-#include "EventRouter.h"
+#include "vapor/FlowParams.h"
+#include "PWidgets.h"
+#include "PFlowRakeRegionSelector.h"
 
 using namespace VAPoR;
+typedef FlowParams FP;
 
-//
-// Register class with object factory!!!
-//
-static RenderEventRouterRegistrar<FlowEventRouter> registrar(
-    FlowEventRouter::GetClassType()
-);
+static RenderEventRouterRegistrar<FlowEventRouter> registrar(FlowEventRouter::GetClassType());
 
-
-FlowEventRouter::FlowEventRouter( QWidget *parent, ControlExec *ce) 
-                    : QTabWidget(parent),
-	                  RenderEventRouter( ce, FlowParams::GetClassType())
+FlowEventRouter::FlowEventRouter(QWidget *parent, ControlExec *ce)
+: RenderEventRouterGUI(ce, FlowParams::GetClassType())
 {
-	_variables = new FlowVariablesSubtab(this);
-	QScrollArea *qsvar = new QScrollArea(this);
-	qsvar->setHorizontalScrollBarPolicy(Qt::ScrollBarAlwaysOff);
-	_variables->adjustSize();
-	qsvar->setWidget(_variables);
-	qsvar->setWidgetResizable(true);
-	addTab(qsvar, "Variables");
-
-	_seeding = new FlowSeedingSubtab(this);
-	QScrollArea* qsseed = new QScrollArea(this);
-	qsseed->setHorizontalScrollBarPolicy(Qt::ScrollBarAlwaysOff);
-	qsseed->setWidget(_seeding);
-	qsseed->setWidgetResizable(true);
-    _seedingTab = qsseed;
-	addTab(qsseed,"Seeding");
-
-	_appearance = new FlowAppearanceSubtab(this);
-	QScrollArea* qsapp = new QScrollArea(this);
-	qsapp->setHorizontalScrollBarPolicy(Qt::ScrollBarAlwaysOff);
-	qsapp->setWidget(_appearance);
-	qsapp->setWidgetResizable(true);
-	addTab(qsapp,"Appearance");
-
-	_geometry = new FlowGeometrySubtab(this);
-	QScrollArea *qsgeo = new QScrollArea(this);
-	qsgeo->setHorizontalScrollBarPolicy(Qt::ScrollBarAlwaysOff);
-	qsgeo->setWidget(_geometry);
-	qsgeo->setWidgetResizable(true);
-	addTab(qsgeo, "Geometry");
+    AddSubtab("Variables", new PGroup({
+        new PSection("Variable Selection", {
+            new S::PDimensionSelector,
+            new S::PXFieldVariableSelector,
+            new S::PYFieldVariableSelector,
+            (new S::PZFieldVariableSelector)->OnlyShowForDim(3),
+            new S::PColorMapVariableSelector,
+        }),
+        new PFidelitySection
+    }));
+    
+    _seedingTab =
+    AddSubtab("Seeding", new PGroup({
+        new PSection("Flow Integration Settings", {
+            new PEnumDropdown(FP::_isSteadyTag, {"Streamlines", "Pathlines"}, {true, false}, "Flow Type"),
+            (new Pif(FP::_isSteadyTag))->Equals(true)->Then({
```

Comment:
Presumably we are only going to use one of these implementations and do away with the other? I.e. before we merge this PR we're going to remove the Pif class use?

Reply:
I don't follow. What are the two implementations?

Reply:
I guess I'm, confused. What is Pif and what does the following do?

(new Pif(FP::_isSteadyTag))->Equals(true)->Then({

Reply:
I added some documentation to the Pif class. It may be beneficial to rename it to `PShowIf`
```
(new Pif("param_tag")->Not->Equals("Hi")->Then(
    PWidgets here will only show if param_tag != "Hi"
)->Else(
    PWidgets here will only show if param_tag == "Hi"
);
```

Reply:
Got it. Yeah, I'd vote for PShowIf

Reply:
Done.